src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -79,23 +79,21 @@ public override ITestEngineRunner MakeTestRunner(TestPackage package)
             // If we have multiple projects or a project plus assemblies
             // then defer to the AggregatingTestRunner, which will make
             // the decision on a file by file basis so that each project
-            // runs with its own settings.
+            // runs with its own settings. Note that bad extensions are
+            // ignored rather than assumed to be projects. This doesn't
+            // really matter since they will result in an error anyway.
             if (projectCount > 1 || projectCount > 0 && assemblyCount > 0)
                 return new AggregatingTestRunner(ServiceContext, package);
 
-            // If we have a single project by itself, make it the top level project.
-            if (projectCount > 0 && assemblyCount == 0)
-                package = package.SubPackages[0];
-
-            // TODO: What about bad extensions?
-
             ProcessModel processModel = GetTargetProcessModel(package);
 
             switch (processModel)
             {
                 default:
                 case ProcessModel.Default:
-                    if (package.SubPackages.Count > 1)
+                    if (projectCount > 0)
+                        return new AggregatingTestRunner(ServiceContext, package);
+                    else if (package.SubPackages.Count > 1 || projectCount > 0)
```

Comment:
The `|| projectCount > 0` will never be true because of the previous if statement. This will still work, but it is redundant.

Reply:
Right! Left over from earlier incarnation. I'll remove it for cleanness.

Reply:
I didn't see that! The `else` is redundant too because of the `return`.